src/explorer/CosmosDBDatabase.ts

```diff
@@ -25,26 +27,39 @@ export class CosmosDBDatabase extends AzureTreeItem<ISiteTreeRoot> {
         };
     }
 
-    // tslint:disable-next-line:function-name
-    public static getLabel(id: string): string {
+    public async deleteTreeItem(): Promise<void> {
+        const connectionToDelete = this.connectionId;
+        const workspaceConfig = vscode.workspace.getConfiguration(constants.extensionPrefix);
+        const connections = workspaceConfig.get<IConnections[]>(constants.configurationSettings.connections, []);
+        const connectionsUnit = connections.find((x: IConnections) => x.webAppId === this.root.client.id);
+        if (connectionsUnit && connectionsUnit.cosmosDB) {
+            const indexToDelete = connectionsUnit.cosmosDB.findIndex((x: string) => x === connectionToDelete);
+            if (indexToDelete > -1) {
+                connectionsUnit.cosmosDB.splice(indexToDelete, 1);
+                workspaceConfig.update(constants.configurationSettings.connections, connections);
+                // tslint:disable-next-line:no-non-null-assertion
+                await this.parent!.refresh();
```

Comment:
You're refreshing the parent node in `extension.ts`, so you don't need to do it here as well. Same feedback for createChildImpl